bedrock/firefox/forms.py

```diff
@@ -5,25 +5,52 @@
 import re
 
 from django import forms
+from django.core.validators import EMPTY_VALUES
+from django.utils.encoding import smart_text
 
-from lib.l10n_utils.dotlang import _
+from lib.l10n_utils.dotlang import _lazy as _
 
 LANG_FILES = ['firefox/whatsnew-fx37']
 
 
+class USPhoneNumberField(forms.CharField):
+    """
+    A form field that validates input as a U.S. phone number.
+    """
+    default_error_messages = {
+        'invalid': _("Sorry. This number isn't valid. Please enter a U.S. phone "
+                     'number or <a href="%s">'
+                     'download directly from Google Play.</a>') % 'http://mzl.la/OgZo6k',
+    }
+
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault('max_length', 14)
+        super(USPhoneNumberField, self).__init__(*args, **kwargs)
+
+    def clean(self, value):
+        super(USPhoneNumberField, self).clean(value)
+        if value in EMPTY_VALUES:
+            return ''
+
+        value = re.sub(r'\D+', '', smart_text(value))
+        if len(value) == 10:
+            value = '1' + value
+        elif len(value) != 11 or value[0] != '1':
+            raise forms.ValidationError(self.error_messages['invalid'])
+
+        return value
+
+
 class SMSSendForm(forms.Form):
-    number = forms.CharField(max_length=14)
+    number = USPhoneNumberField()
     optin = forms.BooleanField(required=False)
 
-    def clean_number(self):
-        mobile = self.cleaned_data['number']
-        mobile = re.sub(r'\D+', '', mobile)
-        if len(mobile) == 10:
-            mobile = '1' + mobile
-        elif len(mobile) != 11 or mobile[0] != '1':
-            raise forms.ValidationError(_(
-                'Sorry. This number isn\'t valid. Please enter a U.S. phone '
-                'number or <a href="%s">'
-                'download directly from Google Play.</a>'
-            ) % ('http://mzl.la/OgZo6k'))
-        return mobile
+
+class SendToDeviceWidgetForm(forms.Form):
+    number = USPhoneNumberField(required=False)
+    email = forms.EmailField(max_length=100, required=False)
+    platform = forms.ChoiceField(choices=(
+        ('ios', 'ios'),
```

Comment:
The actual choices presented to the user will be based on how the widget is used, which will be in android-only mode for now, which means no select box. The submission would technically accept a POST with "ios", but I'm not really concerned about that TBH.